csrc/rsa_gen.cpp

```diff
@@ -21,6 +21,14 @@ JNIEXPORT jlong JNICALL Java_com_amazon_corretto_crypto_provider_RsaGen_generate
     try {
         raii_env env(pEnv);
 
+        // AWS-LC requires that the bitlength be a multiple of 128 and will round down.
+        // We want to guarantee that we return a key of at least the requested strength and so must
+        // round up.
+        jint rounded_bits = bits & ~127;
+        if (rounded_bits < bits) {
+            bits += 128;
+        }
+
         if (FIPS_mode() == 1) {
```

Comment:
It's best to do rounding up in nonFIPS mode only. This would increase the possible values that ACCP allows despite LC not allowing.

Reply:
Is this the consensus solution? I'll happily do it if we think this is what will make us all happy.

Reply:
Yes. It's OK not to throw exception and do rounding up for non-FIPS builds. For FIPS builds, no rounding and throwing exception for the key bit lengths that are not supported.

Reply:
I'm on board with amir's proposal. We can always relax the FIPS strictness if we encounter a legitimate use-case that requires specifying modulus size `% 128 != 0`.